libcnb-cargo/tests/test.rs

```diff
@@ -1,307 +1,330 @@
+            .buildpackage_descriptor
+            .dependencies,
+        expected_buildpackage_dependencies
+    );
 }
 
-impl BuildpackPackagingTest {
-    fn new(fixture_name: &str, target_triple: &str) -> Self {
-        let source_directory = env::current_dir()
-            .unwrap()
-            .join("fixtures")
-            .join(fixture_name);
-        let target_directory = tempdir().unwrap();
-        let copy_options = CopyOptions::new();
-        copy(source_directory, &target_directory, &copy_options).unwrap();
-        BuildpackPackagingTest {
-            fixture_name: fixture_name.to_string(),
-            temp_dir: target_directory,
-            release_build: true,
-            target_triple: String::from(target_triple),
-        }
-    }
+fn create_packaged_buildpack_dir_resolver(
+    cargo_target_dir: &Path,
+    release: bool,
+    target_triple: &str,
+) -> impl Fn(&BuildpackId) -> PathBuf {
+    let cargo_target_dir = PathBuf::from(cargo_target_dir);
+    let target_triple = target_triple.to_string();
 
-    fn dir(&self) -> PathBuf {
-        self.temp_dir
-            .path()
-            .canonicalize()
-            .unwrap()
-            .join(&self.fixture_name)
+    move |buildpack_id| {
+        cargo_target_dir
+            .join("buildpack")
+            .join(&target_triple)
+            .join(if release { "release" } else { "debug" })
+            .join(buildpack_id.as_str().replace('/', "_"))
     }
+}
 
-    fn target_dir_name(&self, buildpack_id: BuildpackId) -> String {
-        self.target_dir(buildpack_id)
-            .canonicalize()
-            .unwrap()
-            .to_string_lossy()
-            .to_string()
-    }
+fn copy_fixture_to_temp_dir(name: &str) -> Result<TempDir, std::io::Error> {
+    let fixture_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+        .join("fixtures")
+        .join(name);
 
-    fn target_dir(&self, buildpack_id: BuildpackId) -> PathBuf {
-        get_buildpack_target_dir(
-            &buildpack_id,
-            &self.dir().join("target"),
-            self.release_build,
-            &self.target_triple,
-        )
-    }
+    env::temp_dir()
+        // libcnb-cargo itself will canonicalize paths too.
+        // If we don't, the string representations of the path won't match. Doing this here will
+        // simplify code at the call-sites since they don't have to canonicalize themselves.
+        // Ref: https://github.com/rust-lang/rust/issues/99608
+        .canonicalize()
+        .and_then(tempdir_in)
+        .and_then(|temp_dir| copy_dir_recursively(&fixture_dir, temp_dir.path()).map(|_| temp_dir))
+}
 
-    fn run_libcnb_package(&self) -> TestOutput {
-        self.run_libcnb_package_from(".")
-    }
+fn copy_dir_recursively(source: &Path, destination: &Path) -> std::io::Result<()> {
+    match fs::create_dir(destination) {
+        Err(io_error) if io_error.kind() == ErrorKind::AlreadyExists => Ok(()),
+        other => other,
+    }?;
```

Comment:
Hm, that could work - yeah. I deliberately didn't use `create_dir_all` because I know that only one path segment should be created by the recursion. If more must be created it see that as an error.

Maybe too pedantic, IDK. I agree that the match isn't the most beautiful code in the world. :)

Reply:
I checked back in Slack, @edmorley is fine keeping this as-is. Considering this resolved.